src/lava/magma/compiler/compiler.py

```diff
@@ -299,13 +299,42 @@ def _create_subcompilers(
             initialized with the Processes they will compile.
         """
         subcompilers = []
+        c_idx = []
+        nc_idx = []
         # Go through all required subcompiler classes...
-        for subcompiler_class, procs in compiler_type_to_procs.items():
+        for idx, (subcompiler_class, procs) in \
+                enumerate(compiler_type_to_procs.items()):
             # ...create the subcompiler instance...
             compiler = subcompiler_class(procs, self._compile_config)
             # ...and add it to the list.
             subcompilers.append(compiler)
-
+            # Remember the index for C and Nc subcompilers:
```

Comment:
If we want to keep the reordering here, please extract this entire reordering into a method with a meaningful method name like `_order_subcompilers_by_type` or something more fitting. This code, using indices, is at too low of an abstraction level to be placed inside of here directly.